api/core/v2/apikey.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+import "github.com/gogo/protobuf@v1.2.1/gogoproto/gogo.proto";
+import "meta.proto";
+
+package sensu.core.v2;
+
+option go_package = "v2";
+option (gogoproto.populate_all) = true;
+option (gogoproto.equal_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.testgen_all) = true;
+
+// An APIKey is an api key specification.
+message APIKey {
+  option (gogoproto.face) = true;
+  option (gogoproto.goproto_getters) = false;
+
+  // Metadata contains the name of the APIKey.
+  ObjectMeta metadata = 1 [(gogoproto.jsontag) = "metadata,omitempty", (gogoproto.embed) = true, (gogoproto.nullable) = false];
+
+  // Key is the API key id.
+  string key = 2;
+
+  // Username is the username associated with the API key.
+  string username = 3;
+
+  // Groups are the groups associated with the API key.
+  repeated string groups = 4;
```

Comment:
I don't think groups should be there. I think only `username` should be there, serving as a "pointer" to an actual user in the store, which itself stores all the relevant details about the user, including their groups. That way if a user gains or loses group memberships, it's automatically reflected in the API keys; no need to keep User and API key groups in sync.

There might be (slight) performance costs to doing that, but I believe it's 100% worth it to have a system that cannot diverge by design.

Reply:
I agree that having both `groups` and `username` defined is a little misleading (following the spec, but unsure why we decided we needed both). If an API key should only be used by one user, than I absolutely agree that we should remove `groups`. However, if we desire that the API key be a little more generically usable, perhaps `groups` is the way to go. Open to discussion, and happy to set a call up to discuss further!

Reply:
Do we rely on the API key groups, or the user groups when determining permissions in RBAC?

1. If we rely on the API key groups, it feels a bit counter intuitive to refer to a user while ignoring its groups.
2. If we rely on the user groups, then we don't need this attribute
3. If we rely on both, it might create some ambiguity and make it hard to understand how a user got permission to whatever resource, and it makes it a bit more complex to remove some permissions (as an operator, I have to remember to modify both the user and its API keys).

TL;DR I might be missing some context about the reason we have a groups attribute here but I feel like it shouldn't probably exist!

Reply:
Removed the groups field (in favor of username) and the key field (in favor of name).